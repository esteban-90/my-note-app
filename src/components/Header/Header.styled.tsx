import styled from '@emotion/styled'
import { bgTransition } from '@/styles'

export const Wrapper = styled.div`
  text-align: center;
  background-color: var(--color-lighter);
  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.1);
  ${bgTransition}

  & h1 {
    margin: 0;
    padding: 0.8rem;
    font-size: 1.7rem;
    color: var(--color-main);
    font-weight: 800;
  }
`
