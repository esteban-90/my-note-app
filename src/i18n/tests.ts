import type { Resource } from 'i18next'
import { createInstance } from 'i18next'
import { initReactI18next } from 'react-i18next'
import { ns, supportedLngs } from './app'

const resources: Resource = {}
export const i18nInstance = createInstance()

void (async () => {
  for (const n of ns) {
    for (const lng of supportedLngs) {
      resources[lng] ??= {}
      resources[lng] = {
        ...resources[lng],
        [n]: await import(`../../public/translations/${lng}/${n}.json`),
      }
    }
  }
})()

i18nInstance.use(initReactI18next).init({
  ns,
  supportedLngs,
  fallbackLng: supportedLngs[0],
  lng: supportedLngs[0],
  react: { useSuspense: false },
  resources,
})
