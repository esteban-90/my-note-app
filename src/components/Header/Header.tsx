import type { FC } from 'react'
import { Wrapper } from './Header.styled'

export const Header: FC = () => {
  return (
    <Wrapper data-testid='app-header'>
      <h1>Note App</h1>
    </Wrapper>
  )
}
