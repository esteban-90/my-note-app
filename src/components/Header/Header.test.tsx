import { screen, render, setName } from '@/tests'
import { Header } from './Header'

describe(setName(Header), () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    void ({ asFragment } = render(<Header />))
  })

  it('should match snapshot', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it(`should render ${Header.name} component`, () => {
    const headerComponent = screen.getByTestId('app-header')
    expect(headerComponent).toBeInTheDocument()
  })

  it('should display the title of the App', () => {
    const appTitle = screen.getByRole('heading', { name: /^note app$/i })
    expect(appTitle).toBeInTheDocument()
  })
})
