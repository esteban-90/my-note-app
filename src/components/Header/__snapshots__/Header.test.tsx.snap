// Vitest Snapshot v1

exports[`HEADER: > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1399pzr"
    data-testid="app-header"
  >
    <h1>
      Note App
    </h1>
  </div>
</DocumentFragment>
`;
